### ISSUE TITLE
Migrate from SSH URI

### ISSUE THREAD

**Issue by NiklasRosenstein** [gogits/gogs#2434]
Is there a way to make it work? Do I have to use a different URI format? I've already generated a keypair for the `gogs` user on my server and added it to GitHub, but that does not seem to be the problem.

![screen shot 2016-01-18 at 19 58 20](https://cloud.githubusercontent.com/assets/1318438/12400027/eea953f6-be1d-11e5-8b90-d703cfb4e808.png)

**Comment by bkcsoft** [comment]
clone it locally on the server running gogs, then "migrate" using `or local server path`

IMO, until the WebUI have support for generating and getting keypairs, this shouldn't be supported in the UI, since it would only be a "half solution" (incomplete)

**Comment by NiklasRosenstein** [comment]
@bkcsoft  I'd be happy without the keypair generator first as I can easily generate a pair myself. But being able to migrate via SSH would've saved me a lot of work.

**Comment by techzilla** [comment]
I have a large amount of repositories on a legacy git server, all using file-system to organize hierarchically.  I also needed a way to have the developers handle their own migrations easily.  

So I wrote a shell script, It's meant to be run from a developer workstation. 
It mirror clones the current git repo, uses the Gogs API to create a matching empty repo , and then does a git push.

I uploaded it to a GitHub Gist, https://gist.github.com/techzilla/85a6ecdd3493606405c5

It's used like,
`./gogs-clonecreatepush.sh -H https://gogs.example.com/ -T ACCESS_TOKEN -O exampleorg git@examplerepo:...`
